TECHFLOW.md

```diff
@@ -0,0 +1,135 @@
+
+The Federal Register does not provide much context when describing CFR
+changes, using asterisks as placeholders for "there is content here". It also
+only provides an indication of what the text is changing _to_, not what it
+was. To figure out this context and more useful change information, eRegs has
+a "diff" feature, which compares two, _complete_ regulation trees and derives
+the difference, which it displays to the user as a "red line" view.
+
+Packaging all of this for notice-and-comment, we need a list of changes pulled
+out from the FR notice (to pull out that `AMDPAR` text and to continue to
+"compile" regulations) as well as version information (i.e. what are the "old"
+and "new" versions of the reg) to pull down the relevant diffs.
+
+_Long term_: Our parser should parse a correct list of changes and derive
+appropriate version information. This data should live with the notice
+structure.
+
+_Present_: The parser knows how to pull of the correct information from
+`AMDPAR`s, generate data necessary for compiling regulations, and how to build
+diffs. It does _not_ know how to figure out what versions of a regulation are
+relevant. Instead, we have a key in a setting, `CFR_CHANGES`, which describes
+the old and new versions per CFR part.
+
+Further, the parser does _not_ know how to package all of this data up. We
+currently need to follow most of the steps in `pipeline` to generate
+appropriate trees and diffs. Unfortunately, this process only account for
+_final_ rules, not proposals. Despite the parser knowing how to generate the
+appropriate amendment information, we don't have a system for
+sending/receiving it from the API. As such, we currently use the `CFR_CHANGES`
+configuration to define something very similar.
+
+### Submission Info
+
+To submit a comment, we need to have a valid docket number (i.e. what document
+to attach this submission to) and a set of fields to request information from
+a submitter. These fields may be drop-down selections, for example limiting
+the user's input to a list of federal agencies. The fields may not all be
+relevant to all users; as a result, we'll likely want to allow some level of
+customization. The fields are generally consistent for a single agency across
+multiple proposals. We should not expect fields to change over the course of a
+comment period.
+
+_Long term_: We can pull down all of this information at parse time and
+include it in the notice structure. The UI would then use the data present in
+this notice structure to send the comments to the correct docket and ask the
+correct questions. Allowing custom templates seems like a good solution for
+the hiding/modifying fields question.
+
+_Present_: We aren't using the notice structure at all. Instead, we have a
+configuration setting for `DOCUMENT_ID` to make this explicit. We have code
+which fetches the appropriate field data from regulations.gov and caches it
+for 24 hours. We don't do anything with customized fields.
```

Comment:
Minor, but (if I'm reading this correctly) we've made a small change on this point since the PR was submitted--we now pass field metadata to the templates so that agency-specific template overrides have access to field options and dependencies.